pkg/apis/v1alpha1/vpcnetworkconfiguration_types.go

```diff
@@ -0,0 +1,116 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// SubnetType describes the subnet type.
+type SubnetType string
+
+const (
+	SubnetTypePublic  SubnetType = "public"
+	SubnetTypePrivate SubnetType = "private"
+)
+
+// VPCNetworkConfigurationSpec defines the desired state of VPCNetworkConfiguration
+type VPCNetworkConfigurationSpec struct {
+	// Namespaces to be applied
+	AppliedToNamespaces []string `json:"appliedToNamespaces,omitempty"`
+	// Tier0 gateway or Tier0 VRF path
+	Tier0Gateway string `json:"tier0Gateway,omitempty"`
+	// Edge cluster paths on which the networking elements will be created
+	// +kubebuilder:validation:Required
```

Comment:
I meant even it is mandatory in NSX-T, we can auto decide an Edge Cluster in WCP. It needs not be optional in this CRD though, as we can let wcpsvc to fill it, but in VC API it could be optional. Could you check the current T1 networking implementation? Is it optional in VC API?

> I remember, in the current T1 networking design, this is optional at least in UI. If not set, we have a algorithm (based on T0 and overlay TZ of the CVDS) to figure out an Edge cluster to use. Could you check if it is optional in VC API or not? If yes, could you investigate what is the right strategy to decide the default ones for a VPC/Project

Reply:
Yes, in VAPI, the fileds are optinal as below for example in workloads.vmodl.

/**
       * NSX Tier0 Gateway used for this namespace.
       * This field does not allow update once applied.
       *
       * @field.optional If {@term unset}, cluster level settings specified in
       * {@link com.vmware.vcenter.namespace_management.Clusters.NCPClusterNetworkInfo#nsxTier0Gateway}
       * will be applied.
       */
      @Resource(ExternalResources.NSX_TIER0_GATEWAY)
      Optional<ID> nsxTier0Gateway;

Reply:
I got your point. So I think we can also set NSXTProject, PublicIPV4Pools required in this CR right? They can be optional in VAPI, but wcpsvc needs to fill value for the required fields.